### PR TITLE
Spl governance integration

### PR DESCRIPTION
**Pull request by adrena-orex**

### CODE REVIEW

**Comment by Adrena-Corto** [comment]
Neat. Can target adx-vest branch

File: programs/perpetuals/src/instructions/add_vest.rs [hunk 1]
```diff
@@ -72,6 +78,38 @@ pub struct AddVest<'info> {
...
+        seeds = [
+            b"lm_token_safe",
+            vest.key().as_ref(),
+        ],
+        token::authority = vest,
+        token::mint = lm_token_mint,
+        bump,
+        payer = payer,
+    )]
+    pub lm_token_safe: Box<Account<'info, TokenAccount>>,
```
Comment (Adrena-Corto) [inline programs/perpetuals/src/instructions/add_vest.rs:92]: nit: Would rename `vest_token_account` (I've kept Alexei nomenclature, I think, for staking too)

He uses "custody_token_account" for Custody, I've made "stake_token_account" for the staking vault/safe

File: programs/perpetuals/src/instructions/add_vest.rs [hunk 2]
```diff
@@ -93,56 +131,125 @@ pub fn add_vest<'info>(
...
+
+        let cpi_program = ctx.accounts.governance_program.to_account_info();
+
+        adapters::deposit_governing_tokens(
+            CpiContext::new(cpi_program, cpi_accounts).with_signer(&[vest_signer_seeds]),
+            ctx.accounts.vest.amount,
+        )?;
+    }
+
+    // Set vote delegate to vest owner
```
Comment (Adrena-Corto) [inline programs/perpetuals/src/instructions/add_vest.rs:225]: really cool <3

File: programs/perpetuals/src/instructions/claim_vest.rs [hunk 3]
```diff
@@ -59,37 +59,110 @@ pub struct ClaimVest<'info> {
...
+        // validate maturation of vest
+        require!(
+            ctx.accounts
+                .vest
+                .is_claimable(ctx.accounts.lm_token_mint.supply)?,
+            PerpetualsError::InvalidVestState
+        );
+    }
+
+    // Revoke vote delegation
```
Comment (Adrena-Corto) [inline programs/perpetuals/src/instructions/claim_vest.rs:103]: ultra-nit: Alexei didn't use any maj in comments 🤣

File: programs/perpetuals/src/instructions/add_vest.rs (hunk 1 above)
Comment (adrena-orex) [reply programs/perpetuals/src/instructions/add_vest.rs]: Good input! Renamed `vest_token_account` ✅

**Pull request merged**